File: iptools/src/main/cpp/ipinfo.cpp
//
// Created by 82039 on 2020/12/15.
//

#include "ipinfo.h"

string ipinfo::getName() {
    return string(name);
}

void ipinfo::setName(string name) {
    memcpy(this->name, name.c_str(), name.length());
}

string ipinfo::getAddress() {
    return string (address);
}

void ipinfo::setAddress(string address) {
    memcpy(this->address, address.c_str(), address.length());
}

string ipinfo::getMac() {
    return string (mac);
}

void ipinfo::setMac(string mac) {
    memcpy(this->mac, mac.c_str(), mac.length());
}

File: iptools/src/main/java/com/jiangc/iptools/IpInfo.java
package com.jiangc.iptools;

public class IpInfo {
    String name;               // 网络接口名称
    String mac;                // 对应mac地址
    String address;            // 对应ip地址

    public IpInfo() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getMac() {
        return mac;
    }

    public void setMac(String mac) {
        this.mac = mac;
    }
}
